x-pack/plugins/security_solution/server/endpoint/routes/resolver/queries/alerts.ts

```diff
@@ -13,19 +14,25 @@ import { JsonObject } from '../../../../../../../../src/plugins/kibana_utils/com
  * Builds a query for retrieving alerts for a node.
  */
 export class AlertsQuery extends ResolverQuery<ResolverEvent[]> {
+  private readonly kqlQuery: JsonObject[] = [];
   constructor(
     private readonly pagination: PaginationBuilder,
     indexPattern: string | string[],
-    endpointID?: string
+    endpointID?: string,
+    kql?: string
   ) {
     super(indexPattern, endpointID);
+    if (kql) {
+      this.kqlQuery.push(esKuery.toElasticsearchQuery(esKuery.fromKueryExpression(kql)));
```

Comment:
The `toElasticsearchQuery` and `fromKueryExpression` can be called on the front end so if we think it's easier to do the conversion there that would work.

Reply:
Think it's fine here, Only benefit I see to moving this to the front end is keeping the backend API a bit slimmer